csci-499-fa24/Team2
# Player Ready Status Toggling with Ready Button

# Pull Request

## Description
Ready button toggles ready status for individual users. Users will not be redirected from waiting room until there are more than one user in the room and all users's statuses are ready.

## What's in this change?

- [x] Added event listeners in `socketClient.js` and `socketServer.js` to handle toggling of user status
- [x] Added player status to player info in socket
- [x] Modified ready button to redirect users after everyone in the room is ready

## Testing changes

To test changes before merging:
- Create a room and join and click the ready button which the user's ready status should be reflected with "ready" and "not ready" next to their name
- Open an incognito account and log into another account and join the same room and upon the first account with a ready status, when the second account toggles ready, both users should be brought to the game board
